spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -512,6 +512,7 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       val newProtocolForLatestMetadata =
         Protocol(readerVersionAsTableProp, writerVersionAsTableProp)
       val proposedNewProtocol = protocolBeforeUpdate.merge(newProtocolForLatestMetadata)
+        .addImplicitLegacyFeatures(newProtocolForLatestMetadata)
```

Comment:
It turns out we already added this capability at some point to `protocol.merge` so this call is not needed at all. However, `downgradeProtocolVersionsIfNeeded` should be called here instead of merge to avoid any surprise downgrades.